### PR TITLE
unsafe: mark unchecked functions as unsafe

### PR DESCRIPTION
**Pull request by vitiral**
Also:
- include interface to get the underlying buffer's pointer
   this can be necessary for allocating/deallocating
- have peek return the reference, improving performance
- remove documentation from peek_unchecked that says it makes "the buffer misbehave" -- it shouldn't. The data might just be invalid

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -84,15 +101,16 @@ impl<'a, T: Clone> CBuf<'a, T>
...
     }
 
     /// Peek next element from the CBuf without removing it
     ///
-    /// Makes the buffer misbehave if it's empty.
+    /// if the buffer is empty, undefined data will be
+    /// returned.
     #[inline]
-    pub fn peek_unchecked(&mut self) -> T {
+    pub fn peek_unchecked(&mut self) -> &T {
```
Comment (dpc) [inline src/lib.rs]: `unsafe` ?

**Comment by dpc** [comment]
LGTM otherwise

File: src/lib.rs (hunk 1 above)
Comment (vitiral) [reply src/lib.rs]: I added docs to it -- it is no more unsafe than having access to a raw pointer (which is not unsafe). The data it returns may be garbage... but that in itself isn't unsafe either.

On the other hand, garbage data could have unintended side-effects, so maybe it should be unsafe. I'll change it now.

Comment (vitiral) [reply src/lib.rs]: changed

**Comment by vitiral** [comment]
should be GTG now

**Pull request merged**